enclave/core/shm.h

```diff
@@ -6,23 +6,23 @@
 
 #include <openenclave/bits/types.h>
 
-typedef struct _shared_memory_pool
+typedef struct _shared_memory_arena
 {
     /* Buffer holding the shared memory pool */
     uint8_t* buffer;
     size_t capacity;
     size_t used;
-    struct _shared_memory_pool* next;
-} Shared_memory_pool;
+    struct _shared_memory_arena* next;
+} Shared_memory_arena;
 
 bool oe_configure_shm_capacity(size_t cap);
```

Comment:
Will rename it. The reason this is not called anywhere currently is because the default arena size 1 mb is sufficient in most cases. But we do need this API to adjust the arena size for functions with gigantic sized parameters, say `char[1000000]`.